### PR TITLE
158619400 edit ride

### PR DESCRIPTION
**Pull request by xcixor**
#### Purpose of the PR
* Create endpoint to edit a ride
#### Tasks to be completed
* Adds functionality to edit a ride
* Creates an endpoint to edit a ride
#### Manual testing
* Navigate to the repo's main page via  [this link](https://github.com/xcixor/my-ride)
* Under the repo's name copy the clone's url
* Open a terminal and navigate to where you want the repo to be downloaded
* Type git clone 'the url you copied above' and press enter to download
* Set up a virtual env and install the dependencies in requirements.txt
* Checkout to the branch 158619400-edit-ride
* run the run.py file to start the server
* Make a request using postman with fields in your ride that you want to edit
#### Running the test
* run the test with pytest like so -> py.test tests/
##### Tests running 
* All rides test
* All controller tests
* All user tests 
##### Tests failing
* Endpoint tests
#### Relevant pivotal tracker stories 
[#158619400](https://www.pivotaltracker.com/n/projects/2179559)
##### ride before edit
ridebefore
#### ride after
rideafter

### CODE REVIEW

File: app/api_1_0/controller.py [hunk 1]
```diff
@@ -43,29 +43,25 @@ def generate_id(item_data, item_id=0):
...
+    #     email = logins.get('Email')
+    #     password = logins.get('Password')
+    #     res = self.user.app_users.get('')
+    #     if res.get('Status'):
+    #         if password == res.get('Message').get('Password'):
+    #             return {'Status': True, 'Message': 'Login Successful'}
+    #         else:
+    #             return {'Status': False, 'Message': 'Password incorrect!'}
+    #     else:
+    #         return {'Status': True, 'Message': 'User does not exist'}
```
Comment (meshack-mbuvi) [inline app/api_1_0/controller.py]: remove code you have commented out

File: app/api_1_0/models.py [hunk 2]
```diff
@@ -57,3 +57,25 @@ def verify_password_length(password):
...
+        """
+        if email in self.app_users:
+            return {'Status': True, 'Message': self.app_users.get(email)}
+        else:
+            return {'Status': False, 'Message': 'That user does not exist'}
+
+    def login(self, email, password):
+        """Check if credentials are correct to allow login."""
+        res = self.get_user(email)
+        print('&&&&&&&&&&&&&&&&&&&&&&', res.get('Message'))
```
Comment (meshack-mbuvi) [inline app/api_1_0/models.py]: remember to remove print statement;It won't be accessible to any user

File: .gitignore [hunk 3]
```diff
@@ -7,3 +7,4 @@ wireframes
 *pytest_cache/*
 *__pycache__*
 *pytest_cache*
```
Comment (meshack-mbuvi) [inline .gitignore:9]: Add .gitignore to .gitignore file., should not be pushed

**Review by meshack-mbuvi — COMMENTED** [review]
Looks fine

**Comment by LWanjiru** [comment]
Why is the Travis build failing? @xcixor

File: .gitignore (hunk 3 above)
Comment (artorious) [reply .gitignore:9]: Good to know @meshack-mbuvi  will take that advice. Thank you.
@xcixor your progress is inspiring

Comment (artorious) [reply .gitignore:9]: Hi @meshack-mbuvi? Question.....if `.gitignore` is not pushed, doesn't that mean that all those who `clone` and `fork` the project need to add one if they're to keep their version clean? Advise.

**Comment by artorious** [comment]
I like what I see. Your progress is inspiring

File: .gitignore (hunk 3 above)
Comment (LWanjiru) [reply .gitignore:9]: @meshack-mbuvi I am curious as to what informs this opinion "Add .gitignore to .gitignore file., should not be pushed" ... Please look into the following resource on this **cc** @artorious  https://help.github.com/articles/ignoring-files/

File: app/api_1_0/controller.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+            user_data = {'Email': email, 'Password': password,
+                         "Type": user_type, 'Confirm Password': confirm_password, 'Id': user_id}
+            result = self.user.create_user(user_data)
+            if result.get('Status'):
+                return {'Status': True, 'Message': result.get('Message')}
+            else:
+                return {'Status': False, 'Message': result.get('Message')}
+
+    @staticmethod
+    def generate_id(item_data, item_id=0):
```
Comment (LWanjiru) [inline app/api_1_0/controller.py]: This function is not explicit, which `list of items` does it cater to? (considering our project is about drivers, passengers and rides)

**Comment by xcixor** [comment]
@LWanjiru , had not fixed the bug I was facing yesterday, fixed it and travis is all good

**Comment by LWanjiru** [comment]
Did you resolve the issues I raised on this PR?

**Comment by xcixor** [comment]
I have fixed all the issues @LWanjiru 
As for codacy it seems to have an issue with previous code that I have already fixed. in the branch the grade is fine

**Review by LWanjiru — CHANGES_REQUESTED** [review]
Some of your checks are failing @xcixor  please look into this

File: app/api_1_0/controller.py (hunk 4 above)
Comment (xcixor) [reply app/api_1_0/controller.py]: I agree it has no relation with the class let fix that

File: .gitignore (hunk 3 above)
Comment (meshack-mbuvi) [reply .gitignore:9]: @LWanjiru Seen and read

File: #1#travis.log [hunk 5]
```diff
@@ -0,0 +1,87 @@
+Command line:
```
Comment (LWanjiru) [inline #1#travis.log]: What does this file do @xcixor ?

File: .codacy.yml [hunk 6]
```diff
@@ -0,0 +1,5 @@
+---
+exclude_paths:
+  - '*.html'
+  - '*.js'
+  - '*.css'
```
Comment (LWanjiru) [inline .codacy.yml]: Add a new line here

File: .travis.yml [hunk 7]
```diff
@@ -1,6 +1,8 @@
 language: python
 python:
  - "3.6.5"
-install: pip install -r requirements.txt
-script: py.test --cov-report --cov app -v
+install:
+- pip install -r requirements.txt
+- pip install coveralls
+script: py.test --cov=app tests/ && coverage report
 after_success: coveralls
```
Comment (LWanjiru) [inline .travis.yml]: add a new line

**Review by LWanjiru — CHANGES_REQUESTED** [review]
Did you set up PYLINT for your project?

File: #1#travis.log (hunk 5 above)
Comment (xcixor) [reply #1#travis.log]: that travis.log is redundant I had been trying a different method of reporting coverage but abandoned it, let me remove that. As for pylint I did set it up and also included it in my requirements.txt

**Comment by xcixor** [comment]
I have fixed the failing tests @LWanjiru they are passing now. I set up pylint on the onset and have trying my best to follow PEP8.

**Comment by LWanjiru** [comment]
Looks good.

**Pull request merged**